File: hello-word/core/api/models/quiz.ts
import { LANG_CODE } from "@/core/constants/common";

export interface QuizResponse {
  id: string;
  questions: QuizQuestion[];
  quizConfig: QuizConfig;
}

export interface QuizConfig {
  totalQuestions: number;
  timePerQuestion?: number;
}

export interface QuizQuestion {
  question: Word;
  answers: Word[];
}

export interface Word {
  id: number;
  categoryId: number;
  [LANG_CODE.EN]: string;
  [LANG_CODE.PL]: string;
}

File: api/internal/data/quiz.go
package data

type QuizWord struct {
	Id         int    `json:"id"`
	CategoryId int    `json:"categoryId"`
	En         string `json:"en"`
	Pl         string `json:"pl"`
}

type QuizQuestion struct {
	Id       int        `json:"id"`
	Question QuizWord   `json:"question"`
	Answers  []QuizWord `json:"answers"`
}

type Quiz struct {
	Id        string         `json:"id"`
	Questions []QuizQuestion `json:"questions"`
}

type GetQuizParam struct {
	QuestionsQty int `json:"questionsQty"`
}
